src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx

```diff
@@ -31,54 +31,39 @@ export const useStakingAction = () => {
     if (wallet) {
       try {
         const adapter = chainAdapterManager.byChain(data.asset.chain)
-        const adapterType = adapter.getType()
 
         let result
 
         const { chainSpecific, validator, action, value } = data
-        const memo = validator === SHAPESHIFT_VALIDATOR_ADDRESS ? 'Delegated with ShapeShift' : ''
+        const memo =
+          validator === SHAPESHIFT_COSMOS_VALIDATOR_ADDRESS ? 'Delegated with ShapeShift' : ''
 
-        if (adapterType === ChainTypes.Cosmos) {
-          switch (action) {
-            case StakingAction.Claim: {
-              result = await (
-                adapter as cosmossdk.cosmos.ChainAdapter
-              ).buildClaimRewardsTransaction({
-                wallet,
-                validator,
-                chainSpecific,
-                memo,
-              })
-              break
-            }
-            case StakingAction.Stake: {
-              result = await (adapter as cosmossdk.cosmos.ChainAdapter).buildDelegateTransaction({
-                wallet,
-                validator,
-                value,
-                chainSpecific,
-                memo,
-              })
-              break
-            }
-            case StakingAction.Unstake: {
-              result = await (adapter as cosmossdk.cosmos.ChainAdapter).buildUndelegateTransaction({
-                wallet,
-                validator,
-                value,
-                chainSpecific,
-                memo,
-              })
-              break
-            }
-            default: {
-              break
-            }
+        const cosmosSdkAdapter = adapter as cosmossdk.cosmos.ChainAdapter
+        const commonArgs = { wallet, validator, chainSpecific, memo }
+        switch (action) {
```

Comment:
I'd love to see us change this to an IIFE instead of mutating `result`, but non-blocking.